d3a_api_client/commands.py

```diff
@@ -47,22 +47,22 @@ def device_info(self, area_uuid):
         return self._add_to_buffer(area_uuid, Commands.DEVICE_INFO, {})
 
     def last_market_stats(self, area_uuid):
-        return self._add_to_buffer(area_uuid, "market_stats", {})
+        return self._add_to_buffer(area_uuid, "market_stats", {"data": {}})
```

Comment:
I think you do not need the "data" part at all here. IIRC, the "data" you probably saw in the d3a code as a key on the payload is added automatically by the requests library IIRC, therefore you should not need this here. Same goes for the rest of the methods too.

Reply:
@spyrostz I couldn't find out why , but sending commands on behalf of A market takes so long and breaks if the `data` key is not sent.
In the same time sending it with tis key works like charm , still investigating anw

Reply:
![Screen Shot 2021-02-17 at 18 48 50](https://user-images.githubusercontent.com/38125715/108237683-d16f8080-7150-11eb-9277-fb9d68970a5f.png)
@spyrostz this is from before the batch commands feature , I'm still not sure why the data key is needed for the market commands and not for the regular devices ones .

Reply:
@spyrostz Found it : 
in `src/d3a/d3a_core/redis_connections/aggregator_connection.py:158` we are loading the data and saving it inside the `pending_batch_commands`
and when processing the command (for example : 
```python
    def market_stats_callback(self, payload):
        market_stats_response_channel = f"{self.channel_prefix}/response/market_stats"
        payload_data = payload["data"] \
            if isinstance(payload["data"], dict) else json.loads(payload["data"])
```
You can see that in here we are trying to get the data key again, this is the root of problem

Reply:
@Hassan754 thanks for taking the time to dig it out! That's okay then all good from my side